src/redux/reducers/role-reducer.js

```diff
@@ -19,10 +20,17 @@ const setLoadingState = state => ({ ...state, isLoading: true });
 const setRecordLoadingState = state => ({ ...state, isRecordLoading: true });
 const setRole = (state, { payload }) => ({ ...state, selectedRole: payload, isRecordLoading: false });
 const setRoles = (state, { payload }) => ({ ...state, roles: payload, isLoading: false });
+const setRolesWithAccess = (state, { payload }) => ({
+  ...state,
+  rolesWithAccess: { ...state.rolesWithAccess, [payload.uuid]: payload },
```

Comment:
You should add the `isLoading` state to `rolesWithAccess` this way you can have more than one row loading at a time.
```JS
{
  rolesWithAccess: {
     [uuid]: {
       isLoadig: true,
       data: payload
     }
  }
}
```

Reply:
Loading multiple rows seems to work fine. To do this I would have do set loading for each uuid in pending action. But in pending action I don't get uuid, only the type. I could probably overcome it somehow, but I don't think it is necessary. Because I am accessing "random" key I am checking for existence of the value anyway.